src/client/api/restful/get_message_metadata.c

```diff
@@ -56,72 +56,44 @@ void msg_meta_free(res_msg_meta_t *res) {
   }
 }
 
-size_t msg_meta_parents_len(res_msg_meta_t *res) {
-  if (res) {
-    if (res->is_error == false) {
-      if (res->u.meta) {
-        return utarray_len(res->u.meta->parents);
-      }
-    }
+size_t msg_meta_parents_count(msg_meta_t *msg) {
+  if (msg) {
+    return utarray_len(msg->parents);
   }
   return 0;
 }
 
-char *msg_meta_parent_get(res_msg_meta_t *res, size_t index) {
-  if (res) {
-    if (index < msg_meta_parents_len(res)) {
-      char **p = (char **)utarray_eltptr(res->u.meta->parents, index);
+char *msg_meta_parent_get(msg_meta_t *msg, size_t index) {
+  if (msg) {
+    if (index < msg_meta_parents_count(msg)) {
+      char **p = (char **)utarray_eltptr(msg->parents, index);
       return *p;
     }
   }
   return NULL;
 }
 
-int msg_meta_deserialize(char const *const j_str, res_msg_meta_t *res) {
-  if (j_str == NULL || res == NULL) {
-    printf("[%s:%d] invalid parameters\n", __func__, __LINE__);
-    return -1;
-  }
-
+int parse_messages_metadata(char const *const j_str, msg_meta_t *res) {
+  int ret = -1;
   cJSON *json_obj = cJSON_Parse(j_str);
   if (!json_obj) {
     printf("[%s:%d]: can not parse JSON object\n", __func__, __LINE__);
     return -1;
   }
-
-  int ret = -1;
-
-  res_err_t *res_err = deser_error(json_obj);
-  if (res_err) {
-    // got an error response
-    res->is_error = true;
-    res->u.error = res_err;
-    ret = 0;
-    goto end;
-  }
-
-  // allocate message metadata object after parsing json object.
-  res->u.meta = metadata_new();
-  if (!res->u.meta) {
-    printf("[%s:%d]: msg_meta_t object allocation failed\n", __func__, __LINE__);
-    goto end;
-  }
-
   // message ID
-  if ((ret = json_get_string_with_prefix(json_obj, JSON_KEY_MSG_ID, res->u.meta->msg_id,
-                                         sizeof(res->u.meta->msg_id))) != 0) {
+  if ((ret = json_get_string_with_prefix(json_obj, JSON_KEY_MSG_ID, res->msg_id, sizeof(res->msg_id))) != 0) {
```

Comment:
`res` object could be NULL, must be checked before accessing it.

Reply:
Done.